docs/translations/translations-in-JS-TS-files.md

```diff
@@ -0,0 +1,108 @@
+# Translations in JS/TS files
+
+In comparison to PHP files, translations in JS/TS files require a few additional steps. First, the Node package [`@wordpress/i18n`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/) needs to be installed:
+
+```sh
+npm install @wordpress/i18n --save
```

Comment:
That's not required if you're within WordPress since you can just have it as a dependency, so this part can be omitted since it depends really on the consumer (and for our case, this is already handled).

Reply:
I do have to admit, that I'm not an expert when it comes to handling translations within JS, but if installing `@wordpress/i18n` is not required, why have we installed it in our package.json: https://github.com/woocommerce/woocommerce-blocks/blob/bf53dd3d98c9d71aff17653cd792a2dec3e1dcb9/package.json#L141

I also wonder why both https://www.npmjs.com/package/@wordpress/i18n and https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/ state that `@wordpress/i18n` has to be installed.

You mentioned `That's not required if you're within WordPress [...]`. Technically, every block lives within WordPress, meaning no extension would have to install that Node package. Yet, there's the npm page and the docs page.

Can you clarify:
- why these documents are there, if installing this dependency is not required?
- why we have it installed, given it's not required?

Reply:
> why we have it installed, given it's not required?

For type hinting.

> state that @wordpress/i18n has to be installed... why these documents are there

Those are standard documents, WordPress doesn't assume you're going to use `dependency-extraction-webpack-plugin`.

For us (WooCommerce Blocks), we use `dependency-extraction-webpack-plugin`, which means webpack will change all `import { x } from '@wordpress/i18n';` to ` const { x } = wp.i18n` and emit `wp-i18n` as a dependency in the `script.assets.php` file.

Also, NPM packages exist for WordPress core, because on release time, WordPress would fetch all of those scripts from webpack and save them locally in `wp-includes` for us to use them, so we rarely need to use it from NPM yourself.

The only time you might use it from npm is within an unit/integration test, but we don't that for translation because it's lot of mocking and hassle to get something to work with low confidence compared to E2E.

Reply:
Ah, I see. I just removed that section from the docs.